cache/file/file.go

```diff
@@ -0,0 +1,52 @@
+// Package file provides a disk based cache.
+package file
+
+import (
+	"github.com/pierrre/imageserver"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+// Cache is a implementation of disk based cache system.
+type Cache struct {
+	Path string
+}
+
+// Get implements imageserver/cache.Cache.
+func (cache *Cache) Get(key string, params imageserver.Params) (*imageserver.Image, error) {
+	data := cache.getData(key)
+	if data == nil {
+		return nil, nil
+	}
+	im := new(imageserver.Image)
+	err := im.UnmarshalBinaryNoCopy(data)
+	if err != nil {
+		return nil, err
+	}
+	return im, nil
+}
+
+func (cache *Cache) getData(key string) []byte {
+	item, err := ioutil.ReadFile(cache.Path + "/" + key)
+	if err != nil {
+		return nil
```

Comment:
We must handle the "file not found" error correctly.
- if the file is not found => return a nil `[]byte` (see https://golang.org/pkg/os/#IsNotExist)
- otherwise return the error